pkg/ingester/ingester_test.go

```diff
@@ -10624,6 +10624,176 @@ func TestIngester_Starting(t *testing.T) {
 	}
 }
 
+func TestIngester_PrepareUnregisterHandler(t *testing.T) {
+	ctx := context.Background()
+
+	overrides, err := validation.NewOverrides(defaultLimitsTestConfig(), nil)
+	require.NoError(t, err)
+
+	type testCase struct {
+		name                     string
+		startIngester            bool
+		httpMethod               string
+		requestBody              io.Reader
+		prepare                  func(i *Ingester)
+		expectedStatusCode       int
+		expectedResponseBody     string
+		expectedUnregisterStatus bool
+	}
+
+	tests := []testCase{
+		{
+			name:                     "returns HTTP 503 if ingester is not running",
+			startIngester:            false,
+			httpMethod:               http.MethodGet,
+			requestBody:              nil,
+			prepare:                  nil,
+			expectedStatusCode:       http.StatusServiceUnavailable,
+			expectedResponseBody:     "",
+			expectedUnregisterStatus: true,
+		},
+		{
+			name:                     "returns HTTP 405 on invalid HTTP method",
+			startIngester:            true,
+			httpMethod:               http.MethodPost,
+			requestBody:              nil,
+			prepare:                  nil,
+			expectedStatusCode:       http.StatusMethodNotAllowed,
+			expectedResponseBody:     "",
+			expectedUnregisterStatus: true,
+		},
+		{
+			name:                     "returns HTTP 400 on GET with request body",
+			startIngester:            true,
+			httpMethod:               http.MethodGet,
+			requestBody:              strings.NewReader(`{"unregister": true}`),
+			prepare:                  nil,
+			expectedStatusCode:       http.StatusBadRequest,
+			expectedResponseBody:     "",
+			expectedUnregisterStatus: true,
+		},
```

Comment:
Such request will not make it to our handler. No need to test for this.

Reply:
:+1:

Action taken: test removed.